app/uk/gov/hmrc/pillar2externalteststub/controllers/SubmitUKTRController.scala

```diff
@@ -43,7 +42,8 @@ class SubmitUKTRController @Inject() (
     with Logging {
 
   def submitUKTR: Action[JsValue] = (Action andThen authFilter).async(parse.json) { implicit request =>
-    request.headers.get("X-Pillar2-Id") match {
+    val plrReference = request.headers.get("X-Pillar2-Id")
```

Comment:
this doesn't seem right. Can you double check please. We should just match on Some(x) and None